File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.py
def dfs(graph, visited, cur):
  visited[cur] = True

  for next in graph[cur]:
    if not visited[next]:
      dfs(graph, visited, next)

def solution():
  n, m = map(int, input().split())
  count = 0
  graph = [[] for _ in range(n + 1)]
  visited = [False for _ in range(n + 1)]
  for _ in range(m):
    a, b = list(map(int, input().split()))
    graph[a].append(b)
    graph[b].append(a)
    
  for i in range(1, n + 1):
    if not visited[i]:
      dfs(graph, visited, i)
      count += 1
  print(count)

solution()

File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.js
let fs = require('fs');
let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const [N, M] = input[0].split(' ').map(Number);

const graph = Array.from({length: N + 1}, () => []);

for (let i = 1; i <= M; i++) {
    const [x, y] = input[i].split(' ').map(Number);
    graph[x].push(y);
    graph[y].push(x);
}

const visited = Array.from({length: N + 1}, () => false);

let answer = 0;

const dfs = (cur) => {
    visited[cur] = true;
    for (const next of graph[cur]) {
        if (!visited[next]) {
            dfs(next);
        }
    }
}

for (let i = 1; i <= N; i++) {
    if(!visited[i]) {
        dfs(i, 0);
        answer += 1;
    }
}

console.log(answer);